### PR TITLE
feat(asm): small refactor of IP retrieval logic + tests

### PR DESCRIPTION
**Pull request by juanjux**
- Don't avoid the loopback IP, consider it like a local one (fix misread RFC).
    - Factorize the global IP checking to a function in compat since 3.x supports `ip_address.is_global`.
    - Add specific tests for IP selection logic between peer and headers sources.
    - Add `config.client_ip_header` instead of os.getenv'ing the env var every time.
    - Remove or replace references to the removed `DD_TRACE_CLIENT_IP_HEADER_DISABLED`.

Signed-off-by: Juanjo Alvarez <juanjo.alvarezmartinez@datadoghq.com>

## Checklist
- [ ] Followed the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) when writing a release note.
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Change contains telemetry where appropriate (logs, metrics, etc.).
- [ ] Telemetry is meaningful, actionable and does not have the potential to leak sensitive data.

### CODE REVIEW

File: ddtrace/contrib/trace_utils.py [hunk 1]
```diff
@@ -240,12 +240,8 @@ def get_header_value(key):  # type: (str) -> Optional[str]
...
         try:
             peer_ip_obj = ipaddress.ip_address(six.text_type(peer_ip))
-            if not peer_ip_obj.is_loopback:
-                # If we have a private_ip and the peer_ip is not public, prefer the one we have
-                if private_ip:
-                    if peer_ip_obj.is_loopback or peer_ip_obj.is_private:
-                        return private_ip
-                # peer_ip is public or we dont have a private_ip to we return the private peer_ip
+            # peer_ip is public or we dont have a private_ip to we return the private peer_ip
+            if not (peer_ip_obj.is_private or peer_ip_obj.is_loopback):
```
Comment (nizox) [inline ddtrace/contrib/trace_utils.py:244]: I think it misses the case when `private_ip` is empty, in this case we should always return the peer ip.

Comment (juanjux) [reply ddtrace/contrib/trace_utils.py]: Not sure it was a problem (the `if tmp_ip_header_value` should avoid empty strings) but I refactored the code a little and added some tests for peer selection including the headers ip being empty, PTAL.

**Review by nizox — APPROVED** [review]

File: tests/tracer/test_trace_utils.py [hunk 2]
```diff
@@ -580,6 +593,95 @@ def test_set_http_meta_headers_ip(
...
+            "invalid",
+            "",
+        ),
+        # Both invalid, empty IP returned
+        (
+            {"x-forwarded-for": "invalid"},
+            "invalid",
+            "",
+        ),
+        # Invalid IP in headers, peer ip should be selected even if loopback
```
Comment (avara1986) [inline tests/tracer/test_trace_utils.py:671]: nice comments :clap:

Comment (juanjux) [reply tests/tracer/test_trace_utils.py:669]: Yeah, wrote them for me mostly since I had to mentally parse the fixtures from the other test above :)

**Review by avara1986 — DISMISSED** [review]

**Review by avara1986 — DISMISSED** [review]

**Review by gnufede — APPROVED** [review]

**Pull request merged**